app/controllers/katello/api/v2/content_view_repositories_controller.rb

```diff
@@ -0,0 +1,39 @@
+module Katello
+  class Api::V2::ContentViewRepositoriesController < Api::V2::ApiController
+    # There are other Content View Repository related endpoints that are in RepositoriesController
+    include Katello::Concerns::FilteredAutoCompleteSearch
+
+    before_action :find_content_view
+    before_action :find_organization_from_cv
+
+    # Shows all repositories, added and available to add, for a content view
+    # Undocumented endpoint since the functionality exists in separate calls already.
+    # This was created for ease of pagination and search for the U
+    param :id, :number, :desc => N_("Content View id"), :required => true
+    def show_all
+      query = Repository.readable
+      ids = @organization.default_content_view.versions.first.repositories.pluck(:id)
+      ids += @content_view.repositories.pluck(:id)
+      query = query.where(id: ids.uniq)
+      sorted_ids = query.sort_by { |repo| repo.in_content_view?(@content_view) ? 0 : 1 }.pluck(:id)
+      sorted_query = Katello::Repository.where(id: sorted_ids)
+
+      # Make sure the added content views are returned first
+      options = { resource_class: Katello::Repository,
+                  deterministic_order: "array_position(array[#{sorted_ids.join(',')}], #{Katello::Repository.table_name}.id)" }
+      repos = scoped_search(sorted_query, nil, nil, options)
+
+      respond_for_index(:collection => repos)
+    end
```

Comment:
I'm not sure if this is the best way to achieve what we are trying to, but we need to ensure that that the added content views are returned first in the results and that stays the same across pagination and regardless of the amount of repositories addded to a CV.

With scoped search, this needs to be done in SQL, but there isn't an attribute to cleanly order by to tell if a repository is added to a specific content view.

This is grabbing all the ids of the available and added repositories to a content view and then sorting them by if they have been added to that content view. From there, queries for all the ids and specifies to use the sql sort by ids in the scoped_search method by the new 'deterministic_order' option. 

This way everything stays searchable and paginated and the added content views will always be on the very top of the results.

Reply:
Can you use a JOIN statement to make a single SQL result set with a simple boolean "added/available" column, and then return that?  It may take some SQL thinking to get it working, and we may have to graduate from the SQL that Active Record can give us, but I think it's at least worth investigating.

Using `JOIN`s will be much better than running `in_content_view?` on every row, because it could turn possibly hundreds of SQL queries into one.

`.uniq` and `.sort_by` can get mighty slow in Ruby when you have a lot of records to process.  `.distinct` and `.order` are much faster since they use SQL.  And you could order first by added, then by ID.  For the `[:only]` params, you could use `.where` clauses to limit the results.